e2e/cli-plugins/run_test.go

```diff
@@ -138,7 +138,7 @@ func TestHelpGood(t *testing.T) {
 	res := icmd.RunCmd(run("-l", "info", "help", "helloworld"))
 	res.Assert(t, icmd.Expected{
 		ExitCode: 0,
-		Err:      icmd.None,
+		Err:      "Warning: Unexpected error, the help command will not have OTEL metrics",
```

Comment:
This should not produce any error as it's considered a "good" way to call help command.

Reply:
Ok fair, so do we want to just avoid instrumenting the help command with otel metrics?

Reply:
To be clear, this warning would only appear if the casting of dockerCli fails

Reply:
Just removing the warning is fine IMO. Otherwise user might run into it when simply making a typo.

Reply:
When exactly does the cast fail? In this test executing `docker run -l info help helloworld` produces the warning.

Reply:
Yeah, it's not `run` (the function name mislead me 🙈). 

But still, I don't think we want to show that warning in case of `docker -l info help helloworld`.

Reply:
should be fixed now

Reply:
Thanks! Just for clarification - this will actually never happen regardless of what user does right? The code would have to change for the type cast to fail?

Reply:
Yep, no user action _should_ be able to break any of these `.(*command.DockerCli)` casts